### PR TITLE
Feat: Adding support for urls in database config

### LINKED ISSUE #304: Add the ability to use a connection url for database config
> A was discussing this with @zacharygolba and wanted to record my thoughts.

> In my opinion, both a full connection string or individual config values (username, password, etc) should be supported for a few reasons. They are both commonly used, depending on the scenario. Connection strings are common when using environment variables such as Heroku, AWS Elastic Beanstalk, or Docker. Lux should support running in a scenario where all configuration is stored in the environment.

> Taking a look at a few well-regarded ORMs, this is a common option:

> http://guides.rubyonrails.org/configuring.html#configuring-a-database
https://hexdocs.pm/ecto/Ecto.html#module-repositories
http://diesel.rs/guides/getting-started/
If you pass a database url, you should not be required to define the database name, or any other database-connection variable. The database url should also take precedence over any other individual connection variable.

> Just my two cents! Cheers.

> \- @aortbals 

--

Related: #289

### PR DESCRIPTION
**Pull request by adampash**
This includes both a url key in the database config and a DATABASE_URL environment variable.

--
Closes #304

### CODE REVIEW

File: src/constants.js [hunk 1]
```diff
@@ -7,4 +7,5 @@ export const CWD = process.cwd();
 export const PID = (isWorker ? worker : process).pid;
 export const PORT = parseInt(ENV.PORT, 10) || 4000;
 export const NODE_ENV = ENV.NODE_ENV || 'development';
+export const DATABASE_URL = ENV.DATABASE_URL;
```
Comment (adampash) [inline src/constants.js]: This is a standard of sorts for PaaS, like Heroku.

File: src/packages/cli/utils/url-message.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  console.log(`
+  You're using a URL in your database config (config/database.js).
+
+  In that case, Lux assumes you don't need to create or drop your database.
+  If you'd like to create or drop a database, set up your database config\
+ without the url.
+
+  For guidance, see:
+  https://github.com/postlight/lux/blob/master/src/packages/database/\
+interfaces.js#L17
```
Comment (adampash) [inline src/packages/cli/utils/url-message.js]: I'm very open to suggestions/edits on this message. Consider this a starting point.

Comment (adampash) [inline src/packages/cli/utils/url-message.js]: At minimum one thing I should probably add is that they could be using a url in the database config or via the DATABASE_URL environment variable.

Comment (zacharygolba) [inline src/packages/cli/utils/url-message.js]: The message is perfect. The only thing I suggest we do is move it to a constant located in `./src/packages/cli/constants.js`.

```javascript
// ./src/packages/cli/constants.js
import template from '../../template';

export const CONNECTION_STRING_MESSAGE = template`
  You're using a URL in your database config (config/database.js).

  In that case, Lux assumes you don't need to create or drop your database.
  If you'd like to create or drop a database, set up your database config\
  without the url.

  For guidance, see:
  https://github.com/postlight/lux/blob/master/src/packages/database/\
  interfaces.js#L17
`;
```

File: src/packages/cli/commands/dbdrop.js [hunk 3]
```diff
@@ -20,6 +22,9 @@ export async function dbdrop() {
   if (driver === 'sqlite3') {
     await rmrf(`${CWD}/db/${database}_${NODE_ENV}.sqlite`);
   } else {
+    if (DATABASE_URL || url) {
+      return urlMessage();
```
Comment (zacharygolba) [inline src/packages/cli/commands/dbdrop.js]: If we change this to a constant we should do the following:

```javascript
if (DATABASE_URL || url) {
  return console.log(CONNECTION_STRING_MESSAGE);
}
```

**Comment by zacharygolba** [comment]
👍 This looks and works great just one comment. Since strings are immutable we can have the `urlMessage` as a constant.

**Pull request merged**